SQRLPlatformAwareInstaller/ViewModels/MainWindowViewModel.cs

```diff
@@ -35,18 +36,23 @@ public ViewModelBase PriorContent
         /// <summary>
         /// Creates a new instance and sets the content of the window.
         /// </summary>
-        public MainWindowViewModel()
+        public MainWindowViewModel(bool rootBail=false)
         {
             ViewModelBase viewModel = null;
 
-            string[] args = Environment.GetCommandLineArgs();
-
-            if (args.Length > 1 && args[1].ToLower() == "-uninstall")
+            if(rootBail)
             {
-                Log.Information($"Installer was called with \"{args[1]}\" command line switch - launching uninstall screen");
-                viewModel = new UninstallViewModel();
+                viewModel = new RootBailViewModel();
             }
             else
+            if (InstallerCommands.Instance != null && InstallerCommands.Instance.Action == InstallerAction.Uninstall)
+            {
+
+
+                Log.Information($"Installer was called with \"{InstallerCommands.Instance}\" command line switches - launching uninstall screen");
+                viewModel = new UninstallViewModel();
+            }
+            else if(InstallerCommands.Instance.Action== InstallerAction.Install)
```

Comment:
Above, we do a `null` check on `InstallerCommands.Instance`.
Is it not required here as well?